### ISSUE TITLE
Support for scikit-learn and other libraries where models are not necessarily differentiable

### ISSUE THREAD

**Issue by drishyamlabs** [interpretml/DiCE#41]
Hi,
Thanks a lot for the wonderful library. As you guys have mentioned in the roadmap, can you tell us by when we can expect the integration of scikit-learn like libraries. I am facing issues when using trying to integrate the same.
Thanks in advance!

**Comment by amit-sharma** [comment]
yes, integrating scikit-learn is a high-priority feature and we are working on it. Should have an update in the next few weeks.

**Comment by arsine1996** [comment]
Hi, thanks for a cool and important library. Can you please tell whether [LightGBM](https://lightgbm.readthedocs.io/en/latest/index.html) will be supported after a new updates?

**Comment by raam93** [comment]
Hi @arsine1996, we are currently trying to make it work for scikit-learn models. We will update the readme if we can extend it to XGBoost and LightGBM as well.

**Comment by konabuta** [comment]
@raam93 I'm also interested in using LightGBM, XGBoost etc. I found currently DiCE supports 'scikit-learn' model [DiCE_model_agnostic_CFs.ipynb](https://github.com/interpretml/DiCE/blob/master/docs/source/notebooks/DiCE_model_agnostic_CFs.ipynb). Does it also support LightGBM scikit-learn API ?

**Comment by amit-sharma** [comment]
@konabuta DiCE supports any ML model that respects the scikit-learn API for `predict` and `predict_proba`. So if your model has these two functions, you can use it with the "sklearn" backend. I think LightGBM uses an API that is compatible with scikit-learn, so I think it should work.  

Just use the backend="sklearn" and input the lightgbm model. It may work with XGBoost too.

For any model, you can have a look at this file: https://github.com/interpretml/DiCE/blob/master/dice_ml/model_interfaces/base_model.py
 As long as your model has a `predict` and `predict_proba` function, it should work directly.

**Comment by jbbqqf** [comment]
Hi — quick triage note. sklearn-family models are first-class today and handled by `dice_ml/model_interfaces/sklearn_model.py` (used via `dice_ml.Model(model=clf, backend='sklearn')`). The `random`, `genetic`, and `kdtree` explainers all work on non-differentiable models — the README's "Quickstart" notebook uses an sklearn pipeline end-to-end. I think this can be closed; reopen with a specific model class that doesn't slot in if you still hit a wall.

---
*This triage comment was prepared with the assistance of Claude (Anthropic). The author reviewed it before posting.*
